transport/packet/src/v1.rs

```diff
@@ -0,0 +1,263 @@
+use std::{
+    fmt::Formatter,
+    ops::{Add, Range, Sub},
+};
+
+/// Tags are represented by 8 bytes ([`u64`]`).
+///
+/// [`u64`] should offer enough space to even avoid collisions and tag attacks.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Tag(pub u64);
+
+impl Tag {
+    pub const MAX: Tag = Tag(u64::MAX);
+    pub const SIZE: usize = size_of::<u64>();
+    pub const USABLE_RANGE: Range<Tag> = ReservedTag::UPPER_BOUND..CustomTag::UPPER_BOUND;
```

Comment:
Docs missing here - what is the difference between `MAX` and `USABLE_RANGE.end` ?